# Analise preço de gasolina no brasil 

Dados retirados do kaggle :https://www.kaggle.com/datasets/matheusfreitag/gas-prices-in-brazil
Esse projeto foi elaborado para fins didaticos onde com auxilio da biblioteca pandas será feito diversas analises no DataFrame.


-Tipos de produtos contidos na base? 
-Gere um DataFrame apenas com gasolina comum
-Qual preço medio de revenda no ano de 2010
-Preço medio de revenda em 2014 na cidade de são paulo 
-Qual a média de preço dos estados da região sul em 2012?
-Qual(quais) estado(s) a gasolina ultrapassou a barreira dos R$ 5,00? E quando isso ocorreu?
-Gerar uma tabela contendo a variação percentual ano a ano para o estado do Rio de Janeiro?


# Importar bibliotecas

In [1]:
import pandas as pd

# Importar banco de dados

In [2]:
dados_2000 = pd.read_csv('gasolina_2000+.csv')
dados_2010 = pd.read_csv('gasolina_2010+.csv')

C:\Users\Gabriel\AppData\Local\Temp\ipykernel_9060\1366514226.py:2: DtypeWarning: Columns (12,14,15,16,17,18) have mixed types. Specify dtype option on import or set low_memory=False.
  dados_2010 = pd.read_csv('gasolina_2010+.csv')


In [3]:
dados_2000.shape


(42638, 19)

In [4]:
dados_2010.shape


(78185, 19)

# Análise 

### Tipo

In [5]:
dados_2000.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 42638 entries, 0 to 42637
Data columns (total 19 columns):
 #   Column                         Non-Null Count  Dtype  
---  ------                         --------------  -----  
 0   Unnamed: 0                     42638 non-null  int64  
 1   DATA INICIAL                   42638 non-null  object 
 2   DATA FINAL                     42638 non-null  object 
 3   REGIÃO                         42638 non-null  object 
 4   ESTADO                         42638 non-null  object 
 5   PRODUTO                        42638 non-null  object 
 6   NÚMERO DE POSTOS PESQUISADOS   42638 non-null  int64  
 7   UNIDADE DE MEDIDA              42638 non-null  object 
 8   PREÇO MÉDIO REVENDA            42638 non-null  float64
 9   DESVIO PADRÃO REVENDA          42638 non-null  float64
 10  PREÇO MÍNIMO REVENDA           42638 non-null  float64
 11  PREÇO MÁXIMO REVENDA           42638 non-null  float64
 12  MARGEM MÉDIA REVENDA           42638 non-null 

In [6]:
dados_2010.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 78185 entries, 0 to 78184
Data columns (total 19 columns):
 #   Column                         Non-Null Count  Dtype  
---  ------                         --------------  -----  
 0   Unnamed: 0                     78185 non-null  int64  
 1   DATA INICIAL                   78185 non-null  object 
 2   DATA FINAL                     78185 non-null  object 
 3   REGIÃO                         78185 non-null  object 
 4   ESTADO                         78185 non-null  object 
 5   PRODUTO                        78185 non-null  object 
 6   NÚMERO DE POSTOS PESQUISADOS   78185 non-null  int64  
 7   UNIDADE DE MEDIDA              78185 non-null  object 
 8   PREÇO MÉDIO REVENDA            78185 non-null  float64
 9   DESVIO PADRÃO REVENDA          78185 non-null  float64
 10  PREÇO MÍNIMO REVENDA           78185 non-null  float64
 11  PREÇO MÁXIMO REVENDA           78185 non-null  float64
 12  MARGEM MÉDIA REVENDA           78185 non-null 

### Unir DataFrames

Como iremos trabalhar com os dados de todo intervalo de tempo, e ambos dataframes tem mesma estrutura, 

iremos unir-los ante de ser feito os tratamentos necessarios

In [7]:
dados_total = pd.concat([dados_2000,dados_2010])

In [8]:
dados_total.shape

(120823, 19)

###  Formatar data

In [9]:
dados_total['DATA FINAL'] = pd.to_datetime(dados_total['DATA FINAL'])
dados_total['ANO'] = dados_total['DATA FINAL'].apply(lambda x :x.year)

### Tipos de produtos ?

In [10]:
dados_total['PRODUTO'].value_counts()

GASOLINA COMUM        23570
GLP                   23561
ETANOL HIDRATADO      23440
ÓLEO DIESEL           21194
GNV                   14469
ÓLEO DIESEL S10        9113
OLEO DIESEL S10        2376
OLEO DIESEL            2351
GASOLINA ADITIVADA      749
Name: PRODUTO, dtype: int64

### DataFrame contendo apenas Gasolina comum

In [11]:
dados = dados_total[dados_total['PRODUTO']=='GASOLINA COMUM'].copy()

In [12]:
dados.shape

(23570, 20)

### Preço medio de revenda no ano de 2010

In [13]:
medio_total = dados['PREÇO MÉDIO REVENDA'].mean()
print(f"Preço medio de revenda total:R$ {medio_total:.2f}")
print(f"Preço medio de revenda ano 2010 foi:R$ {dados[dados['ANO'] == 2010]['PREÇO MÉDIO REVENDA'].mean():.2f}")

Preço medio de revenda total:R$ 3.20
Preço medio de revenda ano 2010 foi:R$ 2.66


### Preço medio de revenda em 2014 na cidade de são paulo

In [14]:
pm = dados[(dados['ESTADO'] == 'SAO PAULO') & (dados['ANO'] == 2014)]['PREÇO MÉDIO REVENDA'].mean()
print(f'O preço medio de revenda em 2014 na cidade de são paulo foi :R$ {pm:.2f}')

O preço medio de revenda em 2014 na cidade de são paulo foi :R$ 2.87


### Qual a média de preço dos estados da região sul em 2012?

In [15]:
pm2 = dados[(dados['REGIÃO']=='SUL')&(dados['ANO']==2012)]['PREÇO MÉDIO REVENDA'].mean()
print(f'O preço medio de revenda dos estados da região sul em 2012 foi :R$ {pm2:.2f}')

O preço medio de revenda dos estados da região sul em 2012 foi :R$ 2.72


### Qual(quais) estado(s) a gasolina ultrapassou a barreira dos R$ 5,00? E quando isso ocorreu?

In [16]:
#se PREÇO MÁXIMO REVENDA > 5,00  mostre ESTADO E ANO
dados[dados['PREÇO MÁXIMO REVENDA']>5][['ESTADO','ANO','PREÇO MÁXIMO REVENDA']].head(20)

ESTADO   ANO  PREÇO MÁXIMO REVENDA
49833            ACRE  2017                 5.100
50133            ACRE  2017                 5.200
50434            ACRE  2017                 5.200
50733            ACRE  2017                 5.200
51033            ACRE  2017                 5.200
51334            ACRE  2017                 5.200
51634            ACRE  2017                 5.200
51935            ACRE  2017                 5.200
52237            ACRE  2017                 5.150
52538            ACRE  2018                 5.150
52551            PARA  2018                 5.040
52556  RIO DE JANEIRO  2018                 5.099
52689            ACRE  2018                 5.150
52702            PARA  2018                 5.040
52840            ACRE  2018                 5.150
52853            PARA  2018                 5.040
52858  RIO DE JANEIRO  2018                 5.099
52992            ACRE  2018                 5.150
53005            PARA  2018                 5.040
53010  RIO DE JANEIRO  2018                 5.099

# Relatório final

### Tipos de produtos contidos na base? 


GASOLINA COMUM        
GLP                  
ETANOL HIDRATADO      
ÓLEO DIESEL           
GNV                   
ÓLEO DIESEL S10        
OLEO DIESEL S10        
OLEO DIESEL            
GASOLINA ADITIVADA      




### Qual preço medio de revenda no ano de 2010

Preço medio de revenda total:R$ 3.20
Preço medio de revenda ano 2010 foi:R$ 2.66

### Preço medio de revenda em 2014 na cidade de são paulo

O preço medio de revenda em 2014 na cidade de são paulo foi :R$ 2.87

### Qual a média de preço dos estados da região sul em 2012?

O preço medio de revenda dos estados da região sul em 2012 foi :R$ 2.72

### Qual(quais) estado(s) a gasolina ultrapassou a barreira dos R$ 5,00? E quando isso ocorreu?

In [18]:
dados[dados['PREÇO MÁXIMO REVENDA']>5][['ESTADO','ANO','PREÇO MÁXIMO REVENDA']].head(20)

ESTADO   ANO  PREÇO MÁXIMO REVENDA
49833            ACRE  2017                 5.100
50133            ACRE  2017                 5.200
50434            ACRE  2017                 5.200
50733            ACRE  2017                 5.200
51033            ACRE  2017                 5.200
51334            ACRE  2017                 5.200
51634            ACRE  2017                 5.200
51935            ACRE  2017                 5.200
52237            ACRE  2017                 5.150
52538            ACRE  2018                 5.150
52551            PARA  2018                 5.040
52556  RIO DE JANEIRO  2018                 5.099
52689            ACRE  2018                 5.150
52702            PARA  2018                 5.040
52840            ACRE  2018                 5.150
52853            PARA  2018                 5.040
52858  RIO DE JANEIRO  2018                 5.099
52992            ACRE  2018                 5.150
53005            PARA  2018                 5.040
53010  RIO DE JANEIRO  2018                 5.099